src/Compilers/CSharp/Portable/Symbols/Wrapped/WrappedLocalSymbol.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a local that is based on another local.
+    /// When inheriting from this class, one shouldn't assume that 
+    /// the default behavior it has is appropriate for every case.
+    /// That behavior should be carefully reviewed and derived type
+    /// should override behavior as appropriate.
+    /// </summary>
+    internal abstract class WrappedLocalSymbol : LocalSymbol
```

Comment:
>WrappedLocalSymbol [](start = 28, length = 18)

It feels like we would want to override Equals methods and GetHashCode in an abstract way.